### PR TITLE
Ensure non-emitting particles not processed on entering tree

### LINKED ISSUE #27066: Weird CPUParticles rendering bug
**Godot version:**
Godot 3.1 RC2, RC3 and stable

**OS/device including version:**
Manjaro Linux

**Issue description:**
There's a very weird CPUParticles rendering bug in Godot 3.1. Some particle effects show a single huge particle, even when they are not emitting. Sometimes it goes away when the partcile effect is emitting for the first time. I haven't yet found out why some particle effects do this and others don't. It's difficult to tell what exactly tiggers this behaviour, but I'll try to create a simple example scene that reproduces it reliably.

I've tried the most recent versions and this bug appears to have been introduced in RC2. It doesn't happen in RC1, which was the last version I tested before the 3.1 release. Sadly the other release candidates came so fast that I didn't have a chance to test them in time.

This is what happens in my game since RC2:
PostRC1

This is what should happen:
RC1

### PR DESCRIPTION
**Pull request by kiidmale**
Also removed non-active CPUParticles(3d) from render list
Fixes #27066

### CODE REVIEW

File: scene/2d/cpu_particles_2d.cpp [hunk 1]
```diff
@@ -965,7 +966,10 @@ void CPUParticles2D::_update_render_thread() {
 void CPUParticles2D::_notification(int p_what) {
 
 	if (p_what == NOTIFICATION_ENTER_TREE) {
-		_set_redraw(true);
+		if (emitting)
+			set_process_internal(true);
+		else
+			set_process_internal(false);
```
Comment (volzhs) [inline scene/2d/cpu_particles_2d.cpp]: this could be simply `set_process_internal(emitting);`

Comment (kiidmale) [reply scene/2d/cpu_particles_2d.cpp]: Ah it really is, will changed the commit :)

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.1.

**Comment by hpvb** [comment]
Reverted the cherry-pick. Causes a regression in tps-demo

**Comment by akien-mga** [comment]
Could be considered against for 3.1.2 together with the regression fix #28380.

**Pull request merged**